### PR TITLE
[IOTDB-4967] Fix SlidingTimeWindow stops when encounter an empty window 

### PR DESCRIPTION
**Pull request by lancelly**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/adapter/ElasticSerializableRowRecordListBackedMultiColumnWindow.java [hunk 1]
```diff
@@ -63,6 +63,9 @@ public int windowSize() {
 
   @Override
   public Row getRow(int rowIndex) throws IOException {
+    if (this.size == 0) {
+      return null;
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/adapter/ElasticSerializableRowRecordListBackedMultiColumnWindow.java:67]: Throw IndexOutOfRangeException is better. Please refer to ArrayList.

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/adapter/ElasticSerializableTVListBackedSingleColumnWindow.java [hunk 2]
```diff
@@ -56,6 +56,9 @@ public int windowSize() {
 
   @Override
   public Row getRow(int rowIndex) {
+    if (this.size == 0) {
+      return null;
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/adapter/ElasticSerializableTVListBackedSingleColumnWindow.java:60]: The same as above.

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/adapter/EmptyRowIterator.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+public class EmptyRowIterator implements RowIterator {
+  @Override
+  public boolean hasNextRow() {
+    return false;
+  }
+
+  @Override
+  public Row next() throws IOException {
+    return null;
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/adapter/EmptyRowIterator.java:35]: Throw InvalidOperationExecption

**Review by SteveYurongSu — COMMENTED** [review]
Minor issues. We can modify the user doc if necessary. And we'd better fix this bug on rel/0.13 branch.

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/adapter/EmptyRowIterator.java (hunk 3 above)
Comment (lancelly) [reply server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/adapter/EmptyRowIterator.java]: done

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/adapter/ElasticSerializableTVListBackedSingleColumnWindow.java (hunk 2 above)
Comment (lancelly) [reply server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/adapter/ElasticSerializableTVListBackedSingleColumnWindow.java]: done

**Comment by lancelly** [comment]
> Minor issues. We can modify the user doc if necessary. And we'd better fix this bug on rel/0.13 branch.

I will fix it on rel/0.13 next week~

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**